app/controllers/flowchart_node_controller.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+class FlowchartNodeController < ApplicationController
+  def create
+    prev_id = params[:prev_id]
+    is_child = params[:is_child]
+    begin
+      prev_node = FlowchartNode.find(prev_id)
+    rescue ActiveRecord::RecordNotFound
+      render status: 404, json: { error: "No node found with id #{prev_id}." }
+    else
+      ActiveRecord::Base.transaction do
+        @new_node = FlowchartNode.create(
+          text: params[:text],
+          header: params[:header],
+          button_text: params[:button_text],
+          next_question: params[:next_question],
+          is_root: false,
+          flowchart_id: prev_node.flowchart_id
+        )
+        if is_child == 'true'
```

Comment:
Why is this checking a string value rather than a boolean in the JSON?

Reply:
Also I don't think this logic needs to be inside of a database transaction, just the saves at the end?

Reply:
I was having issues where the parameter `is_child` always gives me a string, so the check wouldn't work otherwise. Is there a fix for this?

I needed this logic specifically to happen between 2 of the database calls in the transaction.